### DISCUSSION TITLE
Intersect but no Boolean

### DISCUSSION THREAD

**Discussion by ramcdona** [elalish/manifold#556] [General]
Imagine I have MeshA and MeshB.

I want to intersect them such that MeshB leaves its 'impression' on MeshA -- but I don't want any of the parts of MeshB.  In Boolean operations, it would be...

(A-B)+(A^B)

Is there a more direct way to achieve this?

Will it matter?  I.e. since the computational expense is deferred until 'GetMesh()', do N boolean operations cost the same as 1 boolean operation?

Or, will this not work?  I.e. is the system smart enough to optimize away the above operations to just get (A)?

**Comment by pca006132** [comment]
Not sure if there is a better way to achieve this. We do lazy evaluation, but we cannot optimize much in this case.

You can do this slightly more efficiently by using the split function (https://manifoldcad.org/docs/html/classmanifold_1_1_manifold.html#a60e241b8a56d334b3d2d3dbd2fc808ba), which can compute the intersection and difference together. I am not sure if we should add this optimization to csg tree so it is done automatically, this seems a bit complicated and rare.

**Reply by ramcdona** [reply]
So you would propose a Split() followed by a Union()?

Just as an example, I use this to 'mark' a control surface on a wing (among other things).

![Screenshot 2023-09-11 at 11 04 57 AM](https://github.com/elalish/manifold/assets/1329600/25f79c86-50fe-4164-8114-aece52a17816)

**Reply by elalish** [reply]
Yes, I think that's what you want. Our decimator should automatically recognize them as coming from separate instances so that the joints don't get removed in the union. There might be room for a more purpose-built algorithm here someday, but I'd need to understand the use cases and pain points better first.

**Reply by ramcdona** [reply]
Understood.

To make it more complex, I actually do this in a different coordinate system.  The wing exists in XYZ, but I also have parametric UV coordinates from when it was a Bezier surface.

We call this capability a 'SubSurface'.

In order to allow placing a SubSurface on the top -- and to guarantee that it does not peek through to the bottom surface (for example), we actually do SubSurface intersection in UV coordinates.

Right now, both the wing and the subsurface cookie cutter are not Manifold in UV, but I can figure something out there.

I do the cut in UV, then interpolate the XYZ's and then switch the mesh back to XYZ before continuing to intersect with other components.
